src/Agent.Plugins/GitCliManager.cs

```diff
@@ -236,8 +236,46 @@ public async Task<int> GitFetch(AgentTaskPluginExecutionContext context, string
             // add --unshallow to convert the shallow repository to a complete repository
             string depth = fetchDepth > 0 ? $"--depth={fetchDepth}" : (File.Exists(Path.Combine(repositoryPath, ".git", "shallow")) ? "--unshallow" : string.Empty);
 
+            // parse filter and only include valid options
```

Comment:
I think we need a feature flag for this change;; I'll send you the info on how to plumb it though

Reply:
Added `AGENT_USE_FETCH_FILTER_IN_CHECKOUT_TASK`